prow/cluster/trusted_serviceaccounts.yaml

```diff
@@ -31,12 +31,18 @@ metadata:
   name: deployer
   namespace: test-pods
 ---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
```

Comment:
Note that you'll still need to run the gcloud comment to authorize `k8s-artifacts-prod`'s ability to authenticate as `k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com` to GCP.

Reply:
This has already been done by @thockin; I've indepently verified that there is a "Workload Identity" role added to the `k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com` account from the IAM web ui for the `k8s-artifacts-prod` project which owns that GCP SA, so I think we're ready for merge.